tracer/src/Datadog.Trace/Debugger/Sink/AgentBatchUploadApi.cs

```diff
@@ -53,5 +58,71 @@ public async Task<bool> SendBatchAsync(ArraySegment<byte> data)
 
             return true;
         }
+
+        private string BuildUri(string endpoint)
+        {
+            var uri = _apiRequestFactory.GetEndpoint(endpoint);
+            var builder = new UriBuilder(uri);
+            var query = HttpUtility.ParseQueryString(builder.Query);
+            _tags ??= GetDefaultTagsMergedWithGlobalTags();
+            query["ddtags"] = _tags;
+            builder.Query = query.ToString();
+            return builder.ToString();
+        }
+
+        private string GetDefaultTagsMergedWithGlobalTags()
+        {
+            var sb = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+
+            try
+            {
+                var environment = TraceUtil.NormalizeTag(Tracer.Instance.Settings.EnvironmentInternal);
+                if (!string.IsNullOrEmpty(environment))
+                {
+                    sb.Append($"env:{environment},");
+                }
+
+                var version = Tracer.Instance.Settings.ServiceVersionInternal;
+                if (!string.IsNullOrEmpty(version))
+                {
+                    sb.Append($"version:{version},");
+                }
+
+                var hostName = PlatformHelpers.HostMetadata.Instance?.Hostname;
+                if (!string.IsNullOrEmpty(hostName))
+                {
+                    sb.Append($"host:{hostName},");
+                }
+
+                var runtimeId = Tracer.RuntimeId;
+                if (!string.IsNullOrEmpty(runtimeId))
+                {
+                    sb.Append($"{Tags.RuntimeId}:{runtimeId},");
+                }
+
+                var gitRepoUrl = Tracer.Instance.Settings.GitRepositoryUrl;
+                if (!string.IsNullOrEmpty(gitRepoUrl))
+                {
+                    sb.Append($"{Tags.GitRepositoryUrl}:{gitRepoUrl},");
+                }
+
+                var gitCommitSha = Tracer.Instance.Settings.GitCommitSha;
+                if (!string.IsNullOrEmpty(gitCommitSha))
+                {
+                    sb.Append($"{Tags.GitCommitSha}:{gitCommitSha},");
+                }
+
+                foreach (var kvp in Tracer.Instance.Settings.GlobalTagsInternal)
+                {
+                    sb.Append($",{kvp.Key}:{kvp.Value}");
+                }
+
+                return sb.ToString();
+            }
+            finally
+            {
+                StringBuilderCache.Release(sb);
+            }
+        }
```

Comment:
It seems as if you could have tags that are ending with `,` (in case there are no `GlobalTagsInternal`) - is this something we want to avoid?